Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-15E.lua

```diff
@@ -0,0 +1,1315 @@
+
+F_15:definePushButton("F_UFC_KEY_L_GUARD", 56, 3019, 288, "Front UFC", "FRONT Left Guard Receiver - Channel/Manual Key")
+F_15:definePushButton("F_UFC_KEY_A1", 56, 3020, 289, "Front UFC", "FRONT A/1 Key")
+F_15:definePushButton("F_UFC_KEY_N2", 56, 3021, 290, "Front UFC", "FRONT N/2 Key")
+F_15:definePushButton("F_UFC_KEY_B3", 56, 3022, 291, "Front UFC", "FRONT B/3 Key")
+F_15:definePushButton("F_UFC_KEY_R_GUARD", 56, 3023, 292, "Front UFC", "FRONT Right Guard Receiver - Channel/Manual Key")
+F_15:definePushButton("F_UFC_KEY_MARK", 56, 3024, 293, "Front UFC", "FRONT Mark Key")
+F_15:definePushButton("F_UFC_KEY_W4", 56, 3025, 294, "Front UFC", "FRONT W/4 Key")
+F_15:definePushButton("F_UFC_KEY_M5", 56, 3026, 295, "Front UFC", "FRONT M/5 Key")
+F_15:definePushButton("F_UFC_KEY_E6", 56, 3027, 296, "Front UFC", "FRONT E/6 Key")
+F_15:definePushButton("F_UFC_KEY_IP", 56, 3028, 297, "Front UFC", "FRONT I/P Key")
+F_15:definePushButton("F_UFC_KEY_DOT", 56, 3029, 298, "Front UFC", "FRONT Decimal Point Key")
+F_15:definePushButton("F_UFC_KEY_7", 56, 3030, 299, "Front UFC", "FRONT :/7 Key")
+F_15:definePushButton("F_UFC_KEY_S8", 56, 3031, 300, "Front UFC", "FRONT S/8 Key")
+F_15:definePushButton("F_UFC_KEY_C9", 56, 3032, 301, "Front UFC", "FRONT C/9 Key")
+F_15:definePushButton("F_UFC_KEY_SHF", 56, 3033, 302, "Front UFC", "FRONT Shift Key")
+F_15:definePushButton("F_UFC_KEY_AP", 56, 3034, 303, "Front UFC", "FRONT A/P Key")
+F_15:definePushButton("F_UFC_KEY_CLR", 56, 3035, 304, "Front UFC", "FRONT Clear Key")
+F_15:definePushButton("F_UFC_KEY_0", 56, 3036, 305, "Front UFC", "FRONT -/0 Key")
+F_15:definePushButton("F_UFC_KEY_DATA", 56, 3037, 306, "Front UFC", "FRONT Data Key")
+F_15:definePushButton("F_UFC_KEY_MENU", 56, 3038, 307, "Front UFC", "FRONT Menu Key")
+
+local function replaceSpecial(str, result)
+	local special = result or ""
+	local i = str.find(str, "[:.`]")
+
+	if i ~= nil then
+		special = string.format("%s%" .. i - 1 .. "s", special, string.sub(str, i, i))
+		replaceSpecial(string.sub(str, i + 1), special)
+	end
+
+	return special
+end
+
+local function prepareCharsAndSpecial(str)
+	str = str:gsub("°", "`")
+	local strChars = str:gsub("[:.`]", "")
+	local strSpecial = string.len(strChars) > 0 and string.format("%-" .. string.len(strChars) .. "s", replaceSpecial(str)) or ""
+
+	return str, strChars, strSpecial
+end
+
+local function replaceCharAtIndex(str, index, character)
+	local startChars = string.sub(str, 1, index - 1)
+	local remainingChars = string.sub(str, index + 1)
+	return string.format("%s%s%s", startChars, character, remainingChars)
+end
+
+local function replaceIndexIfValue(str, charValue, index)
+	local charData = coerce_nil_to_string(charValue):gsub("°", "`")
+
+	if string.len(charData) > 0 then
+		return replaceCharAtIndex(str, index, charData):gsub("`", "'")
+	end
+
+	return str
+end
+
+local function combine_ufc_line(left, center, right, centerStart)
+	left = coerce_nil_to_string(left)
```

Comment:
Will you be switching to Functions.coerce_nil_to_string?

Reply:
done